### PR TITLE
SW-813 Some products not rendered on category pages v1

### PR DESCRIPTION
**Pull request by Radomir98**

### CODE REVIEW

File: src/Export/Search/AbstractProductSearcher.php [hunk 1]
```diff
@@ -8,6 +8,7 @@
 use FINDOLOGIC\Shopware6Common\Export\Config\PluginConfig;
 use FINDOLOGIC\Shopware6Common\Export\Utils\Utils;
 use InvalidArgumentException;
+use Shopware\Core\Content\Product\Aggregate\ProductVisibility\ProductVisibilityDefinition;
```
Comment (TobiasGraml11) [inline src/Export/Search/AbstractProductSearcher.php:11]: There is no Shopware within the library, this has to be some class from `Vin\ShopwareSdk`

File: src/Export/Search/AbstractProductSearcher.php [hunk 2]
```diff
@@ -155,4 +158,23 @@ abstract protected function getCheapestChild(string $productId): ?ProductEntity;
...
+                if ($productVisibility->salesChannelId === $salesChannelId
+                    && $productVisibility->visibility >= ProductVisibilityDefinition::VISIBILITY_ALL
+                ) {
+                    $isVisible = true;
+                    break;
+                }
+            }
+        }
+
+        return $isVisible;
```
Comment (TobiasGraml11) [inline src/Export/Search/AbstractProductSearcher.php:178]:
```suggestion
        if ($product->active) {
            foreach ($product->visibilities->getElements() as $productVisibility) {
                if ($productVisibility->salesChannelId === $salesChannelId
                    && $productVisibility->visibility >= ProductVisibilityDefinition::VISIBILITY_ALL
                ) {
                    return true;
                }
            }
        }

        return false;
```

File: src/Export/Search/AbstractProductSearcher.php (hunk 1 above)
Comment (TobiasGraml11) [reply src/Export/Search/AbstractProductSearcher.php:11]: Aren't those exactly the ones you are using, or did you add them for the screenshot?

But regarding what exists and what not, we can't use Shopware classes within shopware6-common.

Comment (Radomir98) [reply src/Export/Search/AbstractProductSearcher.php:11]: I just added them for the screenshot, they didn't exist before.
I understand that we cannot use Shopware's classes within shopware6-common.

Comment (TobiasGraml11) [reply src/Export/Search/AbstractProductSearcher.php:11]: If there is no constant, we can either define it ourselves in the class, or just use a hardcoded value

Comment (Radomir98) [reply src/Export/Search/AbstractProductSearcher.php:11]: What do you suggest regarding this matter, or how should we add these constants? Do you perhaps have access to the git repository for `vin-sw` where the `Vin\ShopwareSdk` classes are located (I'm just assuming that we could add them there)?

Comment (Radomir98) [reply src/Export/Search/AbstractProductSearcher.php:11]: Can we add the constant to the `ProductVisibilityDefinition` class located within the `Vin\ShopwareSdk`, or did you mean to add it to the `AbstractProductSearcher` class?

Comment (TobiasGraml11) [reply src/Export/Search/AbstractProductSearcher.php:11]: You can ask for it here: https://github.com/vienthuong/shopware-php-sdk

But for now, I would just put it within the AbstractProductSearcher

**Review by TobiasGraml11 — APPROVED** [review]

**Pull request merged**